extensions/pkg/webhook/controlplane/genericmutator/mutator.go

```diff
@@ -229,16 +230,42 @@ func findFileWithPath(osc *extensionsv1alpha1.OperatingSystemConfig, path string
 }
 
 func (m *mutator) mutateOperatingSystemConfig(ctx context.Context, gctx gcontext.GardenContext, osc, oldOSC *extensionsv1alpha1.OperatingSystemConfig) error {
+	cluster, err := gctx.GetCluster(ctx)
+	if err != nil {
+		return err
+	}
+
+	kubeletVersion, err := semver.NewVersion(cluster.Shoot.Spec.Kubernetes.Version)
+	if err != nil {
+		return err
+	}
+
 	// Mutate kubelet.service unit, if present
 	if content := getKubeletService(osc); content != nil {
-		if err := m.ensureKubeletServiceUnitContent(ctx, gctx, content, getKubeletService(oldOSC)); err != nil {
+		// Deserialize unit options
+		opts, err := m.unitSerializer.Deserialize(*content)
```

Comment:
Why did you choose to parse the unit description string over using a proper API field in the OSC resource?

Reply:
I thought about this approach, though, technically the `OperatingSystemConfig` API is not aware of any such specifics, particularly it is not aware that there might be a `kubelet.service`. This is rather an implementation detail of how gardenlet uses it. In fact, there are `OperatingSystemConfig`s without any `kubelet.service` (think downloader),
so it felt a bit odd to introduce such field. Hence, I didn't want to go down this path (despite from the fact that this would be more code/work).

Reply:
Ok, got it. Another idea:
WDYT about adding a simple file to the OSC under a well-known path, that just contains the kubelet version. This would relief us from plain string parsing.

Reply:
String parsing isn't bad per se  and given that the complexity is very low I would rather proceed as is for now. If you have a strong opinion then I'm open to adapt, of course, but I think this wouldn't bring additional advantages. The unit has to be deserialized anyways, this is coding that was there before.

Reply:
Fine for me, as you prefer.

Reply:
Thanks :)